### ISSUE TITLE
Transmit interrupt stops at random times

### ISSUE THREAD

**Issue by techpaul** [Infineon/XMC-for-Arduino#32]
Have program building up, no external hardware connected so no interference from base XMC100 Boot Kit.

Currently stages of code can be seen from reading random analog values (fine for now will change with hardware attached,) some serial output and controlling of GPIO outputs and delay calls to simplify things for the moment.

When running with serial (to debugger), GPIO mainly outputs for external and board LEDs and delays works fine and can see LEDs on board changing at 1 second delay intervals as expected and messages as expected.

Adding of analogRead causes random time when transmit interrupt stops, then programme eventually gets to spin lock waiting for buffer to empty. This I can tell because the serial output stops appearing at a random time,  but other program steps (changes in LEDs and GPIOs carry on for a while, then LED changes stop showing stuck waiting for serial to catch up.

Comment out the ONE line in a loop that does analogRead and this lock up does not occur
```
// Scan and total on each channel
for( j = 0; j < ADC_AVG_DEPTH; j++ )
   for( i = 0; i <= max; i++ )
     /* adc_readings[ i ] += analogRead( i + A0 )*/;
// Create average
for( i = 0; i <= max; i++ )
   adc_readings[ i ] /= ADC_AVG_DEPTH;
```
So nothing complex, in perfoming 4 scans of 2 channels, then after averaging.

There is some interaction with these types of operation which means transmit interrupt gets missed, or stopped. Not had time to track down further.

**Comment by techpaul** [comment]
This appears to be a poorly added modification in earlier phase.

Disabling TX FIFO in HardwareSerial::begin stops the problem, Programmes that failed before work now.

FIFO had be incorrectly setup and lack of changes to interrupt routine to operate FIFO properly to fill as much FIFO as possible on ONE interrupt, or even checking what FIFO was doing did not help.

The XMCLIB FIFO configure routine relies on bits being set as expected not clearing and setting all the necessary bits. Maybe one day I will get time to do a proper FIFO implementation on RX and TX.

**Comment by techpaul** [comment]
Benn checking through and fix throoughly works on NOT losing tx interrupt (basically getting into a FIFO mess).

Done a bit of tidying up for any sized buffer for serial simpler maths on the fly. Some additional improvements

  Remove some lines of pointless (setting variables to zero that are already zero, unnecessary casts etc)
  Ensure micros() and millis() timer starts at earliest possible moment in main after system_starup code
  Fix serial FIFO setup and tidy ups there 

Pull request in a day or two (I have a life beyond this which some do not believe)

**Comment by techpaul** [comment]
TX interrupt bug and other things fixed in #45

**Comment by techpaul** [comment]
Fix incorporated into V1.1.0 board library
